tests/meta/demultiplex/test_demux_post_processing.py

```diff
@@ -40,107 +41,89 @@ class DemultiplexingScenario(BaseModel):
 
 
 @pytest.mark.parametrize(
-    "scenario",
+    "flow_cell_fixture, sample_ids_fixture",
     [
-        DemultiplexingScenario(
-            flow_cell_directory="flow_cell_directory_name_demultiplexed_with_bcl_convert",
-            flow_cell_name="flow_cell_name_demultiplexed_with_bcl_convert",
-            samples_ids="bcl_convert_demultiplexed_flow_cell_sample_internal_ids",
-        ),
-        DemultiplexingScenario(
-            flow_cell_directory="flow_cell_directory_name_demultiplexed_with_bcl_convert_on_sequencer",
-            flow_cell_name="flow_cell_name_demultiplexed_with_bcl_convert_on_sequencer",
-            samples_ids="bcl_convert_demultiplexed_flow_cell_sample_internal_ids",
-        ),
-        DemultiplexingScenario(
-            flow_cell_directory="flow_cell_directory_name_demultiplexed_with_bcl_convert_flat",
-            flow_cell_name="flow_cell_name_demultiplexed_with_bcl_convert",
-            samples_ids="bcl_convert_demultiplexed_flow_cell_sample_internal_ids",
+        ("novaseq_x_flow_cell", "selected_novaseq_x_sample_ids"),
+        (
+            "novaseq_6000_post_1_5_kits_flow_cell",
+            "selected_novaseq_6000_post_1_5_kits_sample_ids",
         ),
     ],
-    ids=["BCLConvert tree", "BCLConvert on sequencer", "BCLConvert flat"],
+    ids=["BCLConvert on sequencer", "BCLConvert flat"],
 )
 def test_post_processing_of_flow_cell(
-    scenario: DemultiplexingScenario,
-    demultiplex_context: CGConfig,
-    request,
+    flow_cell_fixture: str,
+    sample_ids_fixture: str,
+    updated_demultiplex_context: CGConfig,
+    updated_demux_post_processing_api: DemuxPostProcessingAPI,
     tmp_illumina_demultiplexed_flow_cells_directory: Path,
+    request: pytest.FixtureRequest,
 ):
     """Test adding a demultiplexed flow cell to the databases with. Runs on each type of
     demultiplexing software and setting used."""
 
     # GIVEN a demultiplexed flow cell
-    flow_cell_demultiplexing_directory: str = request.getfixturevalue(scenario.flow_cell_directory)
-    flow_cell_name: str = request.getfixturevalue(scenario.flow_cell_name)
-    sample_internal_ids: list[str] = request.getfixturevalue(scenario.samples_ids)
+    flow_cell: FlowCellDirectoryData = request.getfixturevalue(flow_cell_fixture)
+    flow_cell_demultiplexing_directory: str = flow_cell.full_name
+    flow_cell_name: str = flow_cell.id
```

Comment:
Out of scope but I do not like the name vs id unclarities here.

Reply:
This is an old discussion, see #2264, which I am totally in favour of bringing it back to life, but it is a big refactoring